compiler/rustc_hir/src/definitions.rs

```diff
@@ -220,10 +225,11 @@ impl DefPath {
         write!(s, "::{}", crate_name_str).unwrap();
 
         for component in &self.data {
-            if component.disambiguator == 0 {
-                write!(s, "::{}", component.data.as_symbol()).unwrap();
-            } else {
-                write!(s, "{}[{}]", component.data.as_symbol(), component.disambiguator).unwrap();
```

Comment:
This not printing `::` for the `disambiguator != 0` case looks like a bug, I don't know why that would be.

Reply:
This function (`to_string_friendly`) is not actually used, I'll remove it entirely instead of fixing unused broken code.